lib/puppet/provider/package/gem.rb

```diff
@@ -15,7 +15,14 @@
 
   has_feature :versionable, :install_options, :uninstall_options
 
-  commands :gemcmd => "gem"
+  if Puppet.features.microsoft_windows?
+    # Puppet on Windows prepends its paths to PATH.
+    # One of those paths includes Puppet's vendored ruby and its gem command.
+    # Use a which method that does not traverse those paths.
+    commands :gemcmd => which_without_puppet_paths("gem")
+  else
+    commands :gemcmd => "gem"
+  end
```

Comment:
Just make this entire call use the `commands :gemcmd => which_without_puppet_path("gem")` and handle the non-Windows versus Windows as suggested above.